src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlBuffer.cs

```diff
@@ -288,6 +269,85 @@ internal decimal Decimal
             }
         }
 
+        /// <summary>
+        /// Returns number of trailing zeros using the supplied parameters.
+        /// </summary>
+        /// <param name="data1">An 32-bit unsigned integer which will be combined with data2, data3, and data4</param>
+        /// <param name="data2">An 32-bit unsigned integer which will be combined with data1, data3, and data4</param>
+        /// <param name="data3">An 32-bit unsigned integer which will be combined with data1, data2, and data4</param>
+        /// <param name="data4">An 32-bit unsigned integer which will be combined with data1, data2, and data3</param>
+        /// <param name="scale">The number of decimal places</param>
+        /// <param name="valuablePrecision">OUT |The number of digits without trailing zeros</param>
+        /// <returns>Number of trailing zeros</returns>
+        private static int FindTrailingZerosAndPrec(uint data1, uint data2, uint data3, uint data4, byte scale, out int valuablePrecision)
+        {
+            // Make local copy of data to avoid modifying input.
+            uint[] rgulNumeric = new uint[4] { data1, data2, data3, data4 };
```

Comment:
Can i suggest `Span<uint> rgulNumeric= stackalloc uint[4]{data1, data2, data3, data4};` and theh change `SqlDecimalDivBy` to take a span as well, avoids the array allocation